dask/dataframe/io/parquet/arrow.py

```diff
@@ -119,6 +120,51 @@ def _determine_dataset_parts(fs, paths, gather_statistics, filters, dataset_kwar
     return parts, dataset
 
 
+def _write_partitioned(table, root_path, partition_cols, filesystem=None, **kwargs):
+    """ Write table to a partitioned dataset with pyarrow.
+
+        Logic copied from pyarrow.parquet.
+        (arrow/python/pyarrow/parquet.py::write_to_dataset)
+    """
+    fs, root_path = pq._get_filesystem_and_path(filesystem, root_path)
+    pq._mkdir_if_not_exists(fs, root_path)
+
+    df = table.to_pandas(ignore_metadata=True)
```

Comment:
I don't know to what extent you want to keep the copy of the code from pyarrow as a pure copy, but, if including it here, there are some potential efficiency improvements to be made.

For example, just before calling this method, there is `t = pa.Table.from_pandas(df, preserve_index=preserve_index, schema=schema)` that converts the pandas DataFrame to an arrow table. You can also pass the dataframe directly, avoiding the conversion from arrow table back to a pandas dataframe here.

Reply:
I tried a few different ways of avoiding this.  Everything seems to work fine on my local system when I use the original `df` (with various different environments).  However, the Windows CI is returning [this failure](https://github.com/dask/dask/runs/536672652).  I am having trouble figuring out how to avoid this (especially since I cannot reproduce locally).